src/main/java/org/odessajavaclub/user/adapter/in/rest/UserController.java

```diff
@@ -0,0 +1,91 @@
+package org.odessajavaclub.user.adapter.in.rest;
+
+import lombok.RequiredArgsConstructor;
+import org.odessajavaclub.user.application.port.in.ActivateUserUseCase;
+import org.odessajavaclub.user.application.port.in.CreateUserUseCase;
+import org.odessajavaclub.user.application.port.in.DeactivateUserUseCase;
+import org.odessajavaclub.user.application.port.in.DeleteUserUseCase;
+import org.odessajavaclub.user.application.port.in.GetUserUseCase;
+import org.odessajavaclub.user.application.port.in.GetUsersUseCase;
+import org.odessajavaclub.user.application.port.in.UpdateUserUseCase;
+import org.odessajavaclub.user.domain.User;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/users")
+public class UserController {
+
+    private final CreateUserUseCase createUserUseCase;
+
+    private final GetUsersUseCase getUsersUseCase;
+
+    private final GetUserUseCase getUserUseCase;
```

Comment:
See the book, page 43
```
What About Read-Only Use Cases?
```

Reply:
Renamed to *Query where appropriate

Reply:
As I understand from the book, we can use only one query interface, with different query types inside like this:
```
getUserById
getUsers
getActiveUsers
```

Or am I wrong?

Reply:
Ok, accepted, will aggregate them